feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/input/ui/MonetaryAmountInput.kt

```diff
@@ -0,0 +1,181 @@
+        if (newValue.count { it == decimalSeparator } > 1) return@onValueChange
+        if (newValue.length == 1 && newValue.first() == decimalSeparator) {
+          text = ""
+          return@onValueChange
+        }
+        val numberOfDecimalDigits = newValue.substringAfter(decimalSeparator, missingDelimiterValue = "").length
+        if (numberOfDecimalDigits > maximumFractionDigits) return@onValueChange
+        text = newValue
+        if (newValue.last() != decimalSeparator) {
+          val monetaryAmount = if (newValue.isBlank()) {
+            null
+          } else {
+            MonetaryAmount(
+              newValue.ifBlank { null },
+              currency,
+            )
+          }
+          onInput(monetaryAmount)
+        }
+      },
+      modifier = modifier
+        .fillMaxWidth()
+        .focusRequester(focusRequester),
+      textStyle = LocalTextStyle.current.copy(
+        textAlign = TextAlign.End,
+        color = localContentColor,
+      ),
+      keyboardOptions = KeyboardOptions(
+        autoCorrect = false,
+        keyboardType = if (allowsDecimals(maximumFractionDigits)) KeyboardType.Number else KeyboardType.Decimal,
+        imeAction = ImeAction.Done,
+      ),
+      keyboardActions = KeyboardActions(),
+      singleLine = true,
+      visualTransformation = visualTransformation@{ annotatedString: AnnotatedString ->
+        val transformedString = buildAnnotatedString {
+          val numbersBeforeDecimal = annotatedString.split(decimalSeparator).first()
+          val hasDecimalSeparator = annotatedString.contains(decimalSeparator)
+          val numbersAfterDecimal = annotatedString.split(decimalSeparator).getOrNull(1)
+
+          val numberOfNonTripletNumbers = numbersBeforeDecimal.length % 3
+          append(numbersBeforeDecimal.take(numberOfNonTripletNumbers))
+          if (numberOfNonTripletNumbers != 0 && numbersBeforeDecimal.length > 3) {
+            append(" ")
+          }
+          numbersBeforeDecimal.drop(numberOfNonTripletNumbers).chunked(3).forEachIndexed { index, triplet ->
+            if (index != 0) append(" ")
+            append(triplet)
+          }
+          if (hasDecimalSeparator) {
+            append(decimalSeparator)
+            if (numbersAfterDecimal != null) {
+              append(numbersAfterDecimal)
+            }
+          }
+          append(" ")
+          withStyle(SpanStyle(localContentColor.copy(mediumContentAlpha))) {
+            append(currency)
+          }
+        }
+        TransformedText(
+          transformedString,
+          MonetaryAmountOffsetMapping(annotatedString.text, decimalSeparator),
+        )
+      },
+      cursorBrush = SolidColor(cursorColor),
+      decorationBox = { innerTextField ->
+        BoxWithTrailingIcon(
+          textField = innerTextField,
+          showTrailingIcon = text.isNotEmpty(),
+          onClick = {
+            text = ""
+            focusRequester.freeFocus()
```

Comment:
This does not seem to work. I can only press the "Purchase price"-button (not the input field) once in order to get focus. If I close the keyboard and press the button again, nothing happens.

Reply:
Hmmm I think I've seen smth like this before.
Can you try fetching the manager using
`val focusManager = LocalFocusManager.current`
And then on that do
`focusManager.clearFocus()`
Maybe that helps? Or something like that?

Otherwise not sure, we can figure that out later though.

Reply:
Worked! Thanks